tuxemon/mod_manager/__init__.py

```diff
@@ -86,73 +86,98 @@ def download_package(self, name, release, repo=None, dont_extract=False, install
+            with open(filename, "wb") as file:
+                for chunk in r.iter_content(chunk_size=8096):
+                    file.write(chunk)
+                    downloaded_size += 8096
+                    print(f"{downloaded_size}/{file_size}", int(file_size) - downloaded_size)
 
-        outfolder = os.path.join(paths.BASEDIR, "mods", f"{name}")
+        outfolder = os.path.join(paths.BASEDIR, "mods", name)
 
         self.write_package_to_list(os.path.relpath(outfolder), name)
 
         if not dont_extract:
-            self.extract(filename, outfolder)
-            with open(f"{outfolder}/meta.json", "w") as metafile:
-                meta = self.get_package_info(name, repo)
-                metafile.write(json.dumps(meta, indent=4, sort_keys=False))
+            with zipfile.ZipFile(filename) as zip_:
+                print(zip_.namelist())
+                zip_.extractall(path=os.path.join(paths.BASEDIR, "mods", outfolder))
 
         if install_deps:
             # This function calls download_package, might cause issues
             self.install_dependencies(name, release, repo, dont_extract=dont_extract, done=installed)
 
-    def install_dependencies(self, name, release, repo, dont_extract=False, symlink=True, done=None):
-        """
-        Same as the download_package(), but it includes dependency installing.
-        When symlink is True, dependency's files will be linked.
-        """
-        name = sanitize_paths(name)
-        # Get info
-        meta = self.get_package_info(name, repo)
-
-        installed = done
-        if installed is None:
-            installed = [name]
-        if "dependencies" in meta:
-            dependencies = meta["dependencies"]
-
-            for pack in dependencies:
-                # Sanitize name and release
-                for char in '/\\?%*:|"<>.,;= ':
-                    pack = str(pack).replace(char, "_")
-
-                if pack in installed: continue
-                
-                self.download_package(pack, release, repo, dont_extract=dont_extract, installed=installed)
-
-                # Symlink deps
-                mainfolder = os.path.join(paths.BASEDIR, "mods", name)
-                depfolder = os.path.join(paths.BASEDIR, "mods", pack)
-                symlink_missing(mainfolder, depfolder)
-        else: pass
-
-    def extract(self, file, outfolder):
-        """Extracts the specified zip archive to the mods directory"""
-        with zipfile.ZipFile(file) as zip_:
-            zip_.extractall(path=os.path.join(paths.BASEDIR, "mods", outfolder))
-
-    def get_package_info(self, name, repo):
+    def install_dependencies(self, author, name, repo, symlink=True, **args):
+        """Recursively resolve dependencies and symlink them"""
+        print(author, name, repo)
+        # Request dependencies for specified package
+        r = requests.get(f"{repo}/api/packages/{author}/{name}/dependencies/?only_hard=1")
+        if r.status_code == 404:
```

Comment:
instead for checking for an error, you should be checking for success.  ie, don't check for 404 and quit, check for 200 and continue (or not 200 and then quit)